Kinvey/Kinvey/DataStore.swift

```diff
@@ -2029,20 +2046,9 @@ open class DataStore<T: Persistable> where T: NSObject {
             deviceId: deviceId,
             options: options
         )
-        Promise<RealtimeRouter> { fulfill, reject in
-            do {
-                let realtimeRouter = try self.realtimeRouter()
-                request.execute() { (data, response, error) in
-                    if let response = response, response.isOK {
-                        fulfill(realtimeRouter)
-                    } else {
-                        reject(buildError(data, response, error, self.client))
-                    }
-                }
-            } catch {
-                reject(error)
-            }
-        }.then { realtimeRouter in
+        execute(
```

Comment:
@tejasranade well, `RealtimeRouter` as the name suggest is just a router to map the signals between consumer and producer. The requirement of `subscribe`, `unsubscribe` is out of scope for the router itself.